core/BaseComponent.js

```diff
@@ -370,7 +370,28 @@ export class BaseComponent extends HTMLElement {
       if (this.pauseRender) {
         this.__initCallback();
       } else {
-        this.render();
+        if (this.constructor['__rootStylesLink']) {
+          let root = this.getRootNode();
+          if (!root) {
+            return;
+          }
+          // @ts-ignore
+          let hasLink = root?.querySelector(`link[${DICT.ROOT_STYLE_ATTR_NAME}="${this.constructor.is}"]`);
+          if (hasLink) {
+            this.render();
+            return;
+          }
+          /** @type {HTMLLinkElement} */
+          let rootLink = this.constructor['__rootStylesLink'].cloneNode(true);
+          rootLink.setAttribute(DICT.ROOT_STYLE_ATTR_NAME, this.constructor['is']);
+          rootLink.onload = () => {
+            this.render();
+          };
+          // @ts-ignore
+          root.nodeType === 9 ? root.head.appendChild(rootLink) : root.prepend(rootLink);
```

Comment:
maybe `Node.DOCUMENT_NODE` or comment at least?

```suggestion
          root.nodeType === Node.DOCUMENT_NODE ? root.head.appendChild(rootLink) : root.prepend(rootLink);
```

Reply:
yep